src/PowerShell/PowerShellManagerPool.cs

```diff
@@ -15,37 +20,80 @@ namespace Microsoft.Azure.Functions.PowerShellWorker.PowerShell
     /// </summary>
     internal class PowerShellManagerPool
     {
-        private readonly ILogger _logger;
-        // Today we don't really support the in-proc concurrency. We just hold an instance of PowerShellManager in this field.
-        private PowerShellManager _psManager;
+        private readonly int _upperBound = 25;
+        private readonly MessagingStream _msgStream;
+        private readonly AsyncConcurrentPool<PowerShellManager> _pool;
+        private int _poolSize;
 
         /// <summary>
         /// Constructor of the pool.
         /// </summary>
-        internal PowerShellManagerPool(ILogger logger)
+        internal PowerShellManagerPool(MessagingStream msgStream)
         {
-            _logger = logger;
+            string upperBound = Environment.GetEnvironmentVariable("InProcConcurrencyUpperBound");
+            if (string.IsNullOrEmpty(upperBound) || !int.TryParse(upperBound, out _upperBound))

```

Comment:
Maybe worth logging that we couldn't parse the upperBound if int.TryParse fails. Warning level.

It also might be worth logging that "upperBound was not set - using the default". Verbose/Trace level.